core/contract/src/main/java/org/eclipse/dataspaceconnector/contract/negotiation/AbstractContractNegotiationManager.java

```diff
@@ -136,10 +160,68 @@ public T build() {
             Objects.requireNonNull(manager.executorInstrumentation, "executorInstrumentation");
             Objects.requireNonNull(manager.negotiationStore, "store");
             Objects.requireNonNull(manager.policyStore, "policyStore");
+            Objects.requireNonNull(manager.sendRetryManager, "sendRetryManager");
             manager.commandProcessor = new CommandProcessor<>(manager.commandQueue, manager.commandRunner, manager.monitor);
 
             return manager;
         }
     }
 
+    protected void update(ContractNegotiation negotiation, Consumer<ContractNegotiationListener> observe) {
+        observable.invokeForEach(observe);
+        negotiationStore.save(negotiation);
+    }
+
+    protected void breakLease(ContractNegotiation negotiation) {
+        negotiationStore.save(negotiation);
+    }
+
+    protected class AsyncSendResultHandler {
+        private final String negotiationId;
+        private final String operationDescription;
+        private Consumer<ContractNegotiation> onSuccessHandler;
+        private Consumer<ContractNegotiation> onFailureHandler;
+
+        public AsyncSendResultHandler(String negotiationId, String operationDescription) {
+            this.negotiationId = negotiationId;
+            this.operationDescription = operationDescription;
+        }
+
+        public AsyncSendResultHandler onSuccess(Consumer<ContractNegotiation> onSuccessHandler) {
+            this.onSuccessHandler = onSuccessHandler;
+            return this;
+        }
+
+        public AsyncSendResultHandler onFailure(Consumer<ContractNegotiation> onFailureHandler) {
+            this.onFailureHandler = onFailureHandler;
+            return this;
+        }
+
+        public BiConsumer<Object, Throwable> build() {
+            return (response, throwable) -> {
```

Comment:
`onFailureHandler` and `onSuccessHandler` both can be `null` here. Either defensively add a null-check to the `if`-statements in L209 and L218, or add non-null requirements in the `build` method.
As a third option, both handlers could have default noop consumers.